.tekton/build_and_trigger.yaml

```diff
@@ -49,19 +49,51 @@ spec:
             value: "$(params.repo_url)"
           - name: revision
             value: "$(params.revision)"
+      - name: calculate-tag
+        params:
+          - name: revision
+            value: "{{ revision }}"
+          - name: target_branch
+            value: "{{ target_branch }}"
+          - name: event_type
+            value: "{{ event_type }}"
+        runAfter:
+          - fetch-repo
+        taskSpec:
+          params:
+            - name: revision
+              type: string
+            - name: target_branch
+              type: string
+            - name: event_type
+              type: string
+          results:
+            - name: image_tag
+              description: Image tag to be used by build
+          steps:
+            - name: make-tag
+              image: registry.access.redhat.com/ubi9
+              script: |
+                #!/usr/bin/env bash
+                VERSION_MAJOR=v1
+                if [ "$(params.target_branch)" == "daily" ] || [ "$(params.event_type)" == "push" ]; then
+                  echo "${VERSION_MAJOR}.$(date --utc '+%Y%m%d%H%M%S')" | tee $(results.image_tag.path)
+                else
+                  echo "PR-$(params.revision)" | tee $(results.image_tag.path)
+                fi
       - name: build-image
         workspaces:
           - name: source
             workspace: workspace
         params:
           - name: IMAGE
-            value: quay.io/redhat-appstudio/clair-in-ci:{{revision}}
+            value: "quay.io/redhat-appstudio/clair-in-ci:$(tasks.calculate-tag.results.image_tag)"
```

Comment:
This line causes issues, I don't knwo why.

When I add `{{revision}}` back it works.
Also I don't understand what exactly is wrong, first half of buildah task just works fine and uses value from `image_tag`

Reply:
This is the error, build passed, but task didn't get to push part. You  can also see, that image_tag value has been used there, so I really don't know why it failed later.
```
COMMIT quay.io/redhat-appstudio/clair-in-ci:PR-224ee76d32e10624fb88417d1973c967be682076
--> f92d40dd1f1
Successfully tagged quay.io/redhat-appstudio/clair-in-ci:PR-224ee76d32e10624fb88417d1973c967be682076
f92d40dd1f13f94d375065b6f99b06fce40c5c3c58bec9e8f065b228f50b659b
/tekton/scripts/script-0-rvhs2: line 7: .: filename argument required
.: usage: . filename [arguments]
```